examples/interactions/src/jvmMain/kotlin/ReconnectOnConnectFailExample.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.rsocket.kotlin.*
+import io.rsocket.kotlin.core.*
+import io.rsocket.kotlin.payload.*
+import io.rsocket.kotlin.transport.*
+import io.rsocket.kotlin.transport.local.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+
+@TransportApi
+fun main(): Unit = runBlocking {
+    val server = LocalServer()
+    RSocketServer().bind(server) {
+        RSocketRequestHandler {
+            requestStream {
+                val data = it.data.readText()
+                val metadata = it.metadata?.readText()
+                println("Server received payload: data=$data, metadata=$metadata")
+
+                flow {
+                    repeat(50) { i ->
+                        delay(100)
```

Comment:
delay of a fixed period is nice, but in practice this is probably some exponential backoff with jitter, or strategy that observes external network changes*.  Can this be generalised to some form in a later PR to Flow<Int>?

*On Android you might do something like https://github.com/grpc/grpc-java/blob/master/documentation/android-channel-builder.md

Reply:
I think, that it can be easily implemented both outside rsocket and kotlinx.coroutines, using simple function. But you can create issue here and in kotlinx.coroutines, and we will see, if it's interested there first

Reply:
My point wasn't so much to implement it but to have a pluggable strategy.